VSW-507: Add vul_suc machine

ligallag-amd opened the pull request:
This allows us to develop and test SuC code using qemu. The machine currently has one sockdma device but it should be trivial to add more if needed. stellaris.c is modified because the Zephyr board 'qemu_cortex_m3' supports it so there is no need for us to implement any drivers for it.

## Testing

See testing section of https://github.com/pensando/ainic-suc/pull/2

ivatet-amd commented on hw/arm/stellaris.c:
```diff
@@ -60,6 +61,49 @@ typedef const struct {
+
+static sockdma_info sockdma_infos[] = {
+    {
+    .name = "sockdma-mcdi",
```
nit:
```suggestion
        .name = "sockdma-mcdi",
```

ivatet-amd commented on hw/arm/stellaris.c:
```diff
@@ -1337,6 +1395,19 @@ static void lm3s6965evb_init(MachineState *machine)
+{
+    VulSucMachineState *s = VUL_SUC_MACHINE(machine);
+
+    for(int i = 0; i < ARRAY_SIZE(s->sockdmas); i++) {
```
nit:
```suggestion
    for (int i = 0; i < ARRAY_SIZE(s->sockdmas); i++) {
```
(There are more similar occurrences.)

ivatet-amd reviewed (approved):
Looks good, and sorry for nit-pickery (which is in the spirit of our recent chat in Slack)!

ligallag-amd commented:
force push addresses nitpickery

ligallag-amd commented:
Force push copies the stellaris file and disables the stellaris build. It is easiest to disable this build since both `stellaris.c` and `vul_suc.c` register the same devices which conflict with eachother. If we care enough about this disabled build, we could rename the devices for vul_suc.

ligallag-amd committed:
VSW-507: Add vul_suc machine

This allows us to develop and test SuC code using qemu.

ligallag-amd committed:
VSW-507: Add sockdma mctp device to vul_suc

The machine currently has one sockdma device but it should be trivial to
add more if necessary.

lizzi-xlnx commented on hw/arm/vul_suc.c:
```diff
@@ -0,0 +1,1437 @@
+/* System controller.  */
+
+#define TYPE_STELLARIS_SYS "stellaris-sys"
+OBJECT_DECLARE_SIMPLE_TYPE(ssys_state, STELLARIS_SYS)
```
I think this is why you're getting the type issue. I'd recomment following the vul_model route and do a mass-rename, so that we don't disable the stellaris board.

lizzi-xlnx reviewed (approved):
As already stated on the ticket, I think this is not the ideal board to copy over since it implements (in the board file!!!) many peripherals that aren't needed by the SuC and the virt board would've been a better basis. It's got a better structure and feels easier to extend.

Not blocking on this issue however, since I got here too late and development is already ongoing on this board.
